wgpu-hal/Cargo.toml

```diff
@@ -143,6 +143,8 @@ dx12 = [
     "dep:range-alloc",
     "dep:windows-core",
     "dep:gpu-allocator",
+    "gpu-allocator/d3d12",
```

Comment:
@kpreid is this a problem for feature sets? I'm never quite sure how implicit features work.

Reply:
> I'm never quite sure how implicit features work.

[The rule](https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies) is that if

* a dependency is optional, and
* not mentioned using `dep:` syntax anywhere in `[features]`,

then it gets an implicit feature. (When [RFC 3491](https://rust-lang.github.io/rfcs/3491-remove-implicit-features.html) is implemented in a future edition, implicit features will no longer be created and this will be an error.)